orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/ecs/EcsServerGroupCreator.groovy

```diff
@@ -101,4 +93,73 @@ class EcsServerGroupCreator implements ServerGroupCreator, DeploymentDetailsAwar
       return "$repo:$tag"
     }
   }
+
+  private Artifact getTaskDefArtifact(Stage stage, Object input) {
+    TaskDefinitionArtifact taskDefArtifactInput = mapper.convertValue(input, TaskDefinitionArtifact.class)
+
+    Artifact taskDef = artifactResolver.getBoundArtifactForStage(
+      stage,
+      taskDefArtifactInput.artifactId,
+      taskDefArtifactInput.artifact)
+    if (taskDef == null) {
+      throw new IllegalArgumentException("Unable to bind the task definition artifact");
+    }
+    return taskDef
+  }
+
+  private Map<String, String> getContainerToImageMap(ArrayList<Map<String, Object>> mappings, Stage stage) {
+    def containerToImageMap = [:]
+
+    // each mapping should be in the shape { containerName: "", imageDescription: {}}
+    mappings.each{
+      def imageValue = (Map<String, Object>) it.imageDescription
+      def resolvedImageAddress = getImageAddressFromDescription(imageValue, stage)
+      def name = (String) it.containerName
+      containerToImageMap.put(name, resolvedImageAddress)
+    }
+    return containerToImageMap
+  }
+
+  private String getImageAddressFromDescription(Map<String, Object> description, Stage givenStage) {
+    if (description.fromContext) {
+      if (givenStage.execution.type == ExecutionType.ORCHESTRATION) {
+        // Use image from specific "find image from tags" stage
+        def imageStage = givenStage.findAncestor({
```

Comment:
Merge conflict fail: I think this is supposed to be "ancestorsWithParentPipelines.find" now based on the diff in line 52

Reply:
Hmm seems to be ok now? Not sure if something upstream changed to permit that.